components/org.wso2.carbon.identity.inbound.ldap/src/main/java/org/wso2/carbon/identity/inbound/ldap/impl/ApacheLDAPServer.java

```diff
@@ -0,0 +1,378 @@
+                        "started yet.";
+                log.error(msg);
+                throw new Exception(msg);
+            }
+        } else {
+
+            String msg = "Directory service is null. The LDAP server may not have started yet.";
+            log.error(msg);
+            throw new Exception(msg);
+        }
+
+    }
+
+    @Override
+    public void changeConnectionUserPassword(String password)
+            throws Exception {
+
+        if (this.service != null) {
+            CoreSession adminSession;
+            try {
+                adminSession = this.service.getAdminSession();
+            } catch (Exception e) {
+                String msg = "An error occurred while retraining admin session.";
+                log.error(msg, e);
+                throw new Exception(msg, e);
+            }
+            if (adminSession != null) {
+                LdapPrincipal adminPrincipal = adminSession.getAuthenticatedPrincipal();
+                if (adminPrincipal != null) {
+
+                    String passwordToStore = "{" + ConfigurationConstants.ADMIN_PASSWORD_ALGORITHM +
+                            "}";
+
+                    MessageDigest messageDigest;
+                    try {
+                        messageDigest = MessageDigest.getInstance(
+                                ConfigurationConstants.ADMIN_PASSWORD_ALGORITHM);
+                    } catch (NoSuchAlgorithmException e) {
+                        throw new Exception(
+                                "Could not find digest algorithm - " +
+                                        ConfigurationConstants.ADMIN_PASSWORD_ALGORITHM, e);
+                    }
+                    messageDigest.update(password.getBytes());
+                    byte[] bytes = messageDigest.digest();
+                    String hash = Base64.encode(bytes);
+                    passwordToStore = passwordToStore + hash;
+
+                    adminPrincipal.setUserPassword(passwordToStore.getBytes());
+
+                    EntryAttribute passwordAttribute = new DefaultServerAttribute(
+                            getAttributeType("userPassword"));
+                    passwordAttribute.add(passwordToStore.getBytes());
+
+                    ServerModification serverModification =
+                            new ServerModification(ModificationOperation.REPLACE_ATTRIBUTE,
+                                    passwordAttribute);
+
+                    List<Modification> modifiedList = new ArrayList<Modification>();
+                    modifiedList.add(serverModification);
+
+                    try {
+                        adminSession.modify(adminPrincipal.getClonedName(), modifiedList);
+                    } catch (Exception e) {
+                        String msg = "Failed changing connection user password.";
+                        log.error(msg, e);
+                        throw new Exception(msg, e);
+                    }
```

Comment:
we usually dont put loads of try-catch blocks for a single logic. We usually wrap the entire logic with a try-catch and do the exception handling from there.

Reply:
removed additional try-catch block